File: library/src/main/java/me/shkschneider/skeleton/java/ClassHelper.java
package me.shkschneider.skeleton.java;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import me.shkschneider.skeleton.helper.LogHelper;

public class ClassHelper {

    protected ClassHelper() {
        // Empty
    }

    @Nullable
    public static Class<?> get(@NonNull final String cls) {
        try {
            return Class.forName(cls);
        }
        catch (final ClassNotFoundException e) {
            LogHelper.wtf(e);
            return null;
        }
    }

    public static String canonicalName(@NonNull final Class<?> cls) {
        return cls.getCanonicalName();
    }

    public static String packageName(@NonNull final Class<?> cls) {
        return canonicalName(cls).replaceFirst("\\.[^\\.]+$", "");
    }

    public static String simpleName(@NonNull final Class<?> cls) {
        return cls.getSimpleName();
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/java/ClassHelper.kt
package me.shkschneider.skeleton.java

import me.shkschneider.skeleton.helper.LogHelper

object ClassHelper {

    fun get(cls: String): Class<*>? {
        try {
            return Class.forName(cls)
        } catch (e: ClassNotFoundException) {
            LogHelper.wtf(e)
            return null
        }
    }

    fun canonicalName(cls: Class<*>): String {
        return cls.canonicalName
    }

    fun packageName(cls: Class<*>): String {
        return canonicalName(cls).replaceFirst("\\.[^\\.]+$".toRegex(), "")
    }

    fun simpleName(cls: Class<*>): String {
        return cls.simpleName
    }

}
